### ISSUE TITLE
Can't find gfortan on RHEL 6.4

### ISSUE THREAD

**Issue by ewmoore** [numpy/numpy#5768]
I cannot build Scipy against a current numpy on RHEL 6.4.  Running `gfortran --version` or `gfortran -dumpversion` both provide the same output:
```
GNU Fortran (GCC) 4.4.7 20120313 (Red Hat 4.4.7-3)
Copyright (C) 2010 Free Software Foundation, Inc.

GNU Fortran comes with NO WARRANTY, to the extent permitted by law.
You may redistribute copies of GNU Fortran
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING
```

This has to do with the changes made to how gnu fortran compilers are detected in #5333. Going back to  8a2dd0680e8470aab7ad63622c0d0af703f58d42 works. 

This isn't the newest RHEL, but it is still in the full support phase.

**Comment by rgommers** [comment]
Hmm, the whole point of using ``-dumpversion`` was to remove the need for parsing random version strings. If there's ``gfortran`` versions floating around where ``-dumpversion`` is an alias for ``--version`` we should probably just revert gh-5333.

**Comment by ewmoore** [comment]
Any version of gfortran from earlier than [this gcc commit from Aug 2010](https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=cb22f9309c33b3b166e6c6d3ce9f426ed6c32e5a) will produce the same output for `--version` and `-dumpversion`.  So yes, there absolutely are versions where this is true.  

There is always the try harder option.  Use the new check if `-dumpversion` produces only a single line of output, otherwise try old check against the `--version` output.  This ought to keep the cases fixed by #5333 working and allow older version to still work.

**Comment by rgommers** [comment]
Good idea. Done in gh-5774.
